arrow-ui-data/src/test/kotlin/arrow/ui/PairingTest.kt

```diff
@@ -0,0 +1,39 @@
+package arrow.ui
+
+import arrow.core.ForId
+import arrow.core.Id
+import arrow.core.extensions.id.applicative.applicative
+import arrow.core.extensions.id.monad.monad
+import arrow.core.identity
+import arrow.core.test.UnitSpec
+import arrow.mtl.State
+import arrow.mtl.extensions.functor
+import arrow.mtl.extensions.fx
+import arrow.mtl.fix
+import arrow.ui.extensions.functor
+import io.kotlintest.shouldBe
+
+class PairingTest : UnitSpec() {
+  init {
+
+    "Test Pairing State <-> Store" {
+      val w = Store(0, ::identity)
+      val actions = with (State()) { fx<Int, Unit> {
+          !set(5)
+          !modify { x: Int -> x + 5 }
+          val s = get<Int>()
+          !set(!s * 3 + 1)
+        }.fix()
+      }
+      val w2 = Pairing.pairStateStore<Int>().select(
+        State().functor(),
+        Store().functor(),
+        actions,
+        w.duplicate()
+      ).fix()
+
+      w2.extract() shouldBe 31
```

Comment:
Right now I'm having problems with this test.

It fails because it is unable to cast `Id((A) -> B)` to expected `(A) -> B` at Pairing#pair

```kotlin
fun <A, B, C> pair(FF: Functor<F>, FG: Functor<G>, fa: Kind<F, A>, gb: Kind<G, B>, f: (A, B) -> C): C =
    FF.run {
      FG.run {
        pairing(
          fa.map { a -> a as Any },
          gb.map { b -> b as Any }
        ) { a, b -> f(a as A, b as B) as Any } as C
      }
    }
```

B is expected to be `(A) -> B` but it is `Id((A) -> B)` instead. I'm trying to unwrap `Store.render` but it breaks this function signature because `gb` then is not a higherkind anymore 🤔